lib/modules/manager/puppet/extract.ts

```diff
@@ -0,0 +1,123 @@
+import { logger } from '../../../logger';
+import { parseUrl } from '../../../util/url';
+import { GitTagsDatasource } from '../../datasource/git-tags';
+import { GithubTagsDatasource } from '../../datasource/github-tags';
+import { PuppetForgeDatasource } from '../../datasource/puppet-forge';
+import type { PackageDependency, PackageFile } from '../types';
+import { isGithubUrl, parseGitOwnerRepo } from './common';
+import { parsePuppetfile } from './puppetfile-parser';
+import type { PuppetfileModule } from './types';
+
+function parseForgeDependency(
+  module: PuppetfileModule,
+  forgeUrl: string | null
+): PackageDependency {
+  const dep: PackageDependency = {
+    depName: module.name,
+    datasource: PuppetForgeDatasource.id,
+    packageName: module.name,
+    currentValue: module.version,
+  };
+
+  if (forgeUrl) {
+    dep.registryUrls = [forgeUrl];
+  }
+
+  return dep;
+}
+
+function parseGitDependency(module: PuppetfileModule): PackageDependency {
+  const moduleName = module.name;
+
+  const git = module.tags?.get('git');
+  const tag = module.tags?.get('tag');
+
+  if (!git || !tag) {
+    return {
+      depName: moduleName,
+      gitRef: true,
+      sourceUrl: git,
+      skipReason: 'invalid-version',
+    };
+  }
+
+  const parsedUrl = parseUrl(git);
+  const githubUrl = isGithubUrl(git, parsedUrl);
+
+  if (githubUrl && parsedUrl && parsedUrl.protocol !== 'https:') {
+    logger.warn(
+      `Access to github is only allowed for https, your url was: ${git}`
+    );
+    return {
+      datasource: GithubTagsDatasource.id,
+      depName: moduleName,
+      gitRef: true,
+      sourceUrl: git,
+      skipReason: 'invalid-url',
+    };
+  }
+  const gitOwnerRepo = parseGitOwnerRepo(git, githubUrl);
+
+  if (typeof gitOwnerRepo !== 'string') {
+    // failed to parse git url
+    return {
+      depName: moduleName,
+      ...gitOwnerRepo,
+    };
+  }
+
+  const packageDependency: PackageDependency = {
+    depName: moduleName,
+    packageName: git,
+    repo: git,
+    sourceUrl: git,
+    gitRef: true,
+    currentValue: tag,
+    datasource: GitTagsDatasource.id,
+  };
+
+  if (githubUrl) {
+    packageDependency.packageName = gitOwnerRepo;
+    packageDependency.repo = undefined;
+    packageDependency.githubRepo = gitOwnerRepo;
```

Comment:
remove, invalid properties

Reply:
which invalid property do you mean?

Reply:
```suggestion
```

properties aren't used and shouln'd be set. they are legacy